src/pages/_app.tsx

```diff
@@ -48,15 +91,70 @@ export default function App({ Component, pageProps }: AppProps) {
             setLiffError(error.toString())
           })
       })
+
+    const init = async () => {
+      try {
+        // IMP START - SDK Initialization
+        await web3auth.init();
+        // IMP END - SDK Initialization
+        setProvider(web3auth.provider);
+        console.log(web3auth)
+
+        if (web3auth.connected) {
+          setLoggedIn(true);
+        } else {
+          let web3authProvider
+          console.log(web3auth)
+          if (liff?.isInClient()) {
+            const id_token = liff?.getIDToken()
+            web3authProvider = await web3auth.connectTo(WALLET_ADAPTERS.OPENLOGIN, {
+              loginProvider: "jwt",
+              extraLoginOptions: {
+                id_token: id_token, // in JWT Format
+                verifierIdField: "sub", // same as your JWT Verifier ID
+              },
+            });
+          }
+          web3authProvider = await web3auth.connectTo(WALLET_ADAPTERS.OPENLOGIN, {
+            loginProvider: "line",
+          });
+          setProvider(web3authProvider);
+          if (web3auth.connected) {
+            setLoggedIn(true);
+          }
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    }
+    init()
```

Comment:
As you say, the ordering is not guaranteed. This is probably a bug when isInClient is true.

> You can insert it in line 86 or use liff.ready().

It seems a bit difficult to use ready because there is no stable liff object throughout the function (liffObject can be null).
I changed it to do an insert on line 86. Thank you.